fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -518,18 +522,34 @@ public void exec() throws Exception {
         for (TUniqueId instanceId : instanceIds) {
             profileDoneSignal.addMark(instanceId, -1L /* value is meaningless */);
         }
+
+        deliverExecFragments();
+    }
+
+    private void deliverExecFragments() throws Exception {
+        // Disable pipeline engine for `INSERT INTO`.
+        // TODO: remove this when load supports pipeline engine.
+        boolean enablePipelineEngine = connectContext != null &&
+                connectContext.getSessionVariable().isEnablePipelineEngine() &&
+                fragments.stream().allMatch(PlanFragment::canUsePipeline);
+        boolean enableDeliverBatchFragments = enablePipelineEngine
+                && connectContext.getSessionVariable().isEnableDeliverBatchFragments();
+
+        if (enableDeliverBatchFragments) {
+            deliverExecBatchFragmentsRequests();
```

Comment:
why deliverExecBatchFragmentsRequests needn't enablePipelineEngine. if so. could move enablePipelineEngine logic to deliverExecFragmentRequests

Reply:
`deliverExecBatchFragmentsRequests` always considers `enablePipelineEngine=true`,  because only `enablePipelineEngine=true` will use deliverExecBatchFragmentsRequests for now.
Maybe it's a little confusing, let me add `isEnablePipelineEngine` as  the argument of `deliverExecBatchFragmentsRequests`.